bunsen/bunsen-avro/src/test/java/com/cerner/bunsen/avro/R4AvroConverterTest.java

```diff
@@ -491,7 +492,13 @@ public void testCompile() throws IOException {
 
     // Ensure common types were generated
     for (String fileToBeVerified : filesToBeVerified) {
-      fileToBeVerified = fileToBeVerified.replaceAll("/", File.separator);
+      String fileSeparator = File.separator;
+      // In case of Windows the path should contain `\\` as the file separator (double slash since
+      // java escapes backslash). Replace `\\` with `\\\\` as the regex will eat one backslash.
+      fileSeparator =
+          fileSeparator.replaceAll(
+              Matcher.quoteReplacement("\\"), Matcher.quoteReplacement("\\\\"));
```

Comment:
Sorry, why have you added this extra step? I mean, why not just use `File.separator` in the next `replaceAll`? I saw your comment above but I thought the escaping only matters when we are dealing with constants, e.g., when I want to assign a value to a String (e.g., `String s = "C:\\test"`). But when we are are calling a method (like `replaceAll`) and its argument has a `\` already, it should work as expected; isn't this the case?

In general, I feel a little bit uneasy about our Windows specific code given we can't test it in our dev/CI env, hence trying to keep differences to a minimum.

Reply:
Without the extra step i.e. by directly using the `File.separator` in the `replaceAll` function was failing with the below error. Hence this extra step was added. However, in the latest changes I have refactored the code to use the format `String.join(File.separator, new String[]{"com","cerner"}`.

```java.lang.IllegalArgumentException: character to be escaped is missing

    at java.base/java.util.regex.Matcher.appendExpandedReplacement(Matcher.java:1019)
    at java.base/java.util.regex.Matcher.appendReplacement(Matcher.java:997)
    at java.base/java.util.regex.Matcher.replaceAll(Matcher.java:1181)```

Reply:
I see, this is much better; thank you.